src/config/parse.auto.c

```diff
@@ -10111,6 +10548,7 @@ static const ConfigOption optionResolveOrder[] =
     cfgOptRepoAzureKey,
     cfgOptRepoAzureKeyType,
     cfgOptRepoCipherPass,
+    cfgOptRepoGcsKeyType,
```

Comment:
`gsc-bucket`, `gcs-key`, and `gcs-endpoint` all depend on `gcs-key-type`. So that's why the first three go to the bottom. The list is traversed all the way through in alpha order each time while dependencies are resolved, so things may not end up right next to each other in the list, i.e. the first three are output in the pass after `gcs-key-type` so anything that is alphabetically after `gcs-key-type` in the same pass will get output before those first three. It doesn't matter as long as 1) we can resolve all options in one pass 2) all dependencies of an option are parsed before we get to that option. We must traverse the entire list so there's no benefit to making the code that does the ordering any more complex.